### PR TITLE
feat: brings back --make-paths-enum option to generate ApiPaths enum

### PR DESCRIPTION
**Pull request by laurenz-glueck**
## Changes

Brings back `--make-paths-enum` option to generate `ApiPaths` enum.
See: https://github.com/openapi-ts/openapi-typescript/issues/1057

## Checklist

- [x] Unit tests updated
- [x] `docs/` updated (if necessary)
- [ ] `pnpm run update:examples` run (only applicable for openapi-typescript)

### CODE REVIEW

**Comment by gzm0** [comment]
@drwpow it seems you have some more context on this.

I suspect the [discussion](https://github.com/openapi-ts/openapi-typescript/issues/1057#issuecomment-1875994876) in the issue went missing because it was closed.

**Comment by drwpow** [comment]
Sorry I think this was just an accidental omission for 7.x, but since it’s an opt-in feature flag I support adding it! (I’ll only try to ensure that default behavior doesn’t change).

I’m in favor of this merging, but we’ll need to add a few tests just to ensure this feature doesn’t break. Also please add a `minor` changeset for this feature so we can release it 🙏

**Comment by laurenz-glueck** [comment]
@drwpow Sounds good!
Please give me a few days and I will try to add some tests and also the changeset…

**Comment by laurenz-glueck** [comment]
@drwpow I have added a few tests and a minor changeset.
Please let me know what you think.

File: docs/cli.md [hunk 1]
```diff
@@ -207,3 +208,22 @@ This results in more explicit typechecking of array lengths.
...
 
 _Thanks, [@kgtkr](https://github.com/kgtkr)!_
+
+### makePathsEnum
+
+This option is useful for generating an enum for all paths in the schema. This can be useful to use the paths from the schema in your code.
+
+Enabling `--make-paths-enum` will add an `ApiPaths` enum like this to the generated types:
+
+::: code-group
```
Comment (drwpow) [inline docs/cli.md:218]: Great docs, thank you!

File: packages/openapi-typescript/test/transform/paths-enum.test.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            get: {},
+          },
+        },
+        want: `export enum ApiPaths {
+    GetApiV1User = "/api/v1/user"
+}`,
+      },
+    ],
+    [
+      "basic with path parameter",
```
Comment (drwpow) [inline packages/openapi-typescript/test/transform/paths-enum.test.ts:23]: Great tests! 🎉

**Review by drwpow — APPROVED** [review]
This is great, thank you! All tests pass and current behavior doesn’t change when the flag isn’t enabled. Will get this merged and shipped!

**Pull request merged**